idx/memory/memory.go

```diff
@@ -1826,6 +1826,68 @@ ORGS:
 	return pruned, nil
 }
 
+// AddDef adds def to the index.
+func (m *UnpartitionedMemoryIdx) AddDefs(defs []schema.MetricDefinition) {
```

Comment:
is it deliberate that the archive.LastSave field is not getting set?

Reply:
In the normal "write to cassandra paths" `CasIdx.{AddOrUpdate,Update}()` we update LastSave (after adding to the write queue) to whatever `uint32(time.Now().Unix())` was before we put it in the write queue.

my suggestion is to do the same in `CasIdx.AddDefs` and `BigtableIdx.AddDefs`